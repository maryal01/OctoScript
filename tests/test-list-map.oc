list<int> l1 = [1, 2, 3, 4, 5];
list<bool> l2 = l1.map(fn (int x) => {x % 2 == 0});
print(string_of_list(l2));

list<float> l3 = [1.1, 2.2, 3.3, 4.4, 5.5];
list<string> l4 = l3.map(fn (float x) => {if (x > 3.) ? "yes" : "no"});
print(string_of_list(l4));

list<bool> l5 = [true, true, false, true, false];
list<float> l6 = l5.map(fn (bool x) => {if (x) ? 1. : 2.});
print(string_of_list(l6));

list<string> l7 = ["aaa", "bb", "ccccc", "dddd", "ee"];
list<int> l8 = l7.map(fn (string x) => {strlen(x)});
print(string_of_list(l8));

