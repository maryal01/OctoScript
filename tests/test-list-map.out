[false, true, false, true, false][no, no, yes, yes, yes][1.000000, 1.000000, 2.000000, 1.000000, 2.000000][3, 2, 5, 4, 2]